packages/output/src/browser/output-widget.tsx

```diff
@@ -147,6 +147,11 @@ export class OutputWidget extends BaseWidget implements StatefulWidget {
         }
     }
 
+    protected onAfterShow(msg: Message): void {
+        super.onAfterShow(msg);
+        this.onResize(Widget.ResizeMessage.UnknownSize); // Triggers an editor widget resize. (#8361)
```

Comment:
I wonder what is the difference to explicit `this.editor.refresh();`.

In `EditorWidget` we do it, we do it as well for `onAfterAttach` to resize it if a widget gets moved between areas.

Reply:
Yes, I started by calling `editor.refresh()` but it did not fix the problem.

Reply:
`this.autoSizing` is `false` in the monaco editor, so calling `refresh` is a NOOP.

Reply:
It turned out, calling `editor.refresh()` does nothing at all when invoked from `EditorWidget#onAfterShow`. `autoSizing` is always `false`.

Reply:
> In `EditorWidget` we do it, we do it as well for `onAfterAttach` to resize it if a widget gets moved between areas.

I have tried opening, maximizing, and DNDing editors while having breakpoints on both `onAfterAttach` and `onAfterShow` in the editor widget:

https://github.com/eclipse-theia/theia/blob/565a23c6d7e1ed763704897730a8760faf57dc09/packages/editor/src/browser/editor-widget.ts#L63

https://github.com/eclipse-theia/theia/blob/565a23c6d7e1ed763704897730a8760faf57dc09/packages/editor/src/browser/editor-widget.ts#L69

Neither of them has any effect as the underlying monaco editor's `autoresize` is a noop.

Reply:
Maybe, somehow related to the old issue? https://github.com/microsoft/monaco-editor/issues/794